src/GenotypeView.js

```diff
@@ -191,6 +227,52 @@ class GenotypeView extends Component {
         });
     }
 
+    _fetchConsequence() {
+        const assembly_info = find(
+            this._resolveAssemblyList(this.state.speciesAssemblyMap, this.state.selectedSpecies),
+            (item => item.speciesAssemblyCode === this.state.selectedAssembly)
+        )
+        if (assembly_info.assemblyAccession !== this.state.ensemblAssembly){
+            console.log(`Cannot fetch consequences because Ensembl does not support ${assembly_info.assemblyAccession}`)
+            return
+        }
+        const ensemblSpecies = assembly_info.taxonomyScientificName.toLowerCase().replace(' ', '_')
+        const inputVariantsMap = Object.fromEntries(this.state.variants.map(
+            item => [[item.chromosome, item.start, '.', item.reference, item.alternate, '. . .'].join(' '), item]
+        ));
+        const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ variants: keys(inputVariantsMap) })
+        };
+        fetch(`https://rest.ensembl.org/vep/${ensemblSpecies}/region`, requestOptions)
+            .then(response => response.json())
+            .then(variantsconsequences => {
+                variantsconsequences.map(consequence => {
+                    const original_variant = inputVariantsMap[consequence.input];
+                    const consequenceType = consequenceTypeDetails[consequence.most_severe_consequence]
+                    if (consequenceType){
+                        original_variant.color = consequenceType.color;
+                    }else{
+                        console.log(`Missing definition for consequence ${consequence.most_severe_consequence}`)
+                    }
+
+                });
+                this.setState({variants:Object.values(inputVariantsMap)});
+            });
+    }
+
+    _fetchConsequenceRanking() {
+        this.setState({isLoaded: false});
+        fetch(`https://rest.ensembl.org/info/variation/consequence_types?content-type=application/json&rank=1`)
+            .then(response => response.json())
+            .then(consequences => {
+                this.consequenceRankMap = Object.fromEntries(
+                    consequences.map(consequence => [consequence.SO_accession, parseInt(consequence.consequence_ranking)])
+                );
+            });
+    }
+
```

Comment:
Thinking a bit about code organisation, since I agree with your concern about the complexity of this class - would it help to split out the data fetching functionality into separate classes, something like `EvaManager` to fetch data from EVA and `EnsemblManager` to fetch data from Ensembl?

To me this is the most obvious next step for refactoring, but maybe there's some complications I'm not seeing...

Reply:
I'll create anothe ticket for that.